testutil/metrictest/assertions.go

```diff
@@ -0,0 +1,236 @@
+	ipRE        = regexp.MustCompile(`[0-9\.]+`)
+	authorityRE = regexp.MustCompile(`[a-zA-Z\-]+\.[a-zA-Z\-]+\.svc\.cluster\.local:[0-9]+`)
+)
+
+// HasNoOutboundHTTPRequest returns error if there is any
+// series matching request_total{direction="outbound"}
+func HasNoOutboundHTTPRequest(metrics, ns string) error {
+	m := promm.NewMatcher("request_total", promm.Labels{
+		"direction": promm.Equals("outbound"),
+	})
+	ok, err := m.HasMatchInString(metrics)
+	if err != nil {
+		return fmt.Errorf("failed to run a check of against the provided metrics: %w", err)
+	}
+	if ok {
+		return fmt.Errorf("expected not to find HTTP outbound requests \n%s", metrics)
+	}
+	return nil
+}
+
+// HasOutboundHTTPRequestWithTLS checks there is a series matching:
+// request_total{
+//   direction="outbound",
+//   target_addr=~"[0-9\.]+:[0-9]+",
+//   target_ip=~"[0-9.]+",
+//   tls="true",
+//   dst_namespace="default.${ns}.serviceaccount.identity.linkerd.cluster.local",
+//   dst_serviceaccount="default"
+// }
+func HasOutboundHTTPRequestWithTLS(metrics, ns string) error {
+	m := promm.NewMatcher("request_total", promm.Labels{
+		"direction":          promm.Equals("outbound"),
+		"target_addr":        promm.Like(addrRE),
+		"target_ip":          promm.Like(ipRE),
+		"tls":                promm.Equals("true"),
+		"server_id":          promm.Equals(fmt.Sprintf("default.%s.serviceaccount.identity.linkerd.cluster.local", ns)),
+		"dst_namespace":      promm.Equals(ns),
+		"dst_serviceaccount": promm.Equals("default"),
+	},
+		promm.HasPositiveValue())
+	ok, err := m.HasMatchInString(metrics)
+	if err != nil {
+		return fmt.Errorf("failed to run a check of against the provided metrics: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf("expected to find HTTP outbound requests \n%s", metrics)
+	}
+	return nil
+}
+
+// HasOutboundHTTPRequestNoTLS checks there is a series matching:
+// request_total{
+//   direction="outbound",
+//   target_addr=~"[0-9\.]+:[0-9]+",
+//   target_ip=~"[0-9.]+",
+//   tls="no_identity",
+//   no_tls_reason="not_provided_by_service_discovery",
+//   dst_namespace="default.${ns}.serviceaccount.identity.linkerd.cluster.local",
+//   dst_serviceaccount="default"
+// }
+func HasOutboundHTTPRequestNoTLS(metrics, ns string) error {
+	m := promm.NewMatcher("request_total", promm.Labels{
+		"direction":          promm.Equals("outbound"),
+		"target_addr":        promm.Like(addrRE),
+		"target_ip":          promm.Like(ipRE),
+		"tls":                promm.Equals("no_identity"),
+		"no_tls_reason":      promm.Equals("not_provided_by_service_discovery"),
+		"dst_namespace":      promm.Equals(ns),
+		"dst_serviceaccount": promm.Equals("default"),
+	},
+		promm.HasPositiveValue())
+	ok, err := m.HasMatchInString(metrics)
+	if err != nil {
+		return fmt.Errorf("failed to run a check of against the provided metrics: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf("expected to find HTTP outbound requests \n%s", metrics)
```

Comment:
The error message here is very similar to the TLS counterpart, should we perhaps differentiate it somehow? I'm thinking it might make debugging a bit easier when tests mess up. wdyt?

Reply:
TBH error messages is one of the hardest things here. :-) I did something in https://github.com/linkerd/linkerd2/pull/8547/commits/17f149d164d1f12ecc04e7745fe014caa1ec1218 but if you have more ideas, I can apply them as well.

Reply:
100%. Think the non-TLS/TLS distinction you added in is more than enough to help us unpack any ambiguity.